src/controllers/todo_record_controller.py

```diff
@@ -91,12 +94,37 @@ def PUT(self, todo_description=None):
             res_msg['data'] = 'Item not found.'
         return res_msg
 
+
+    @cherrypy.expose
     @cherrypy.tools.json_out()
-    def DELETE(self, record_id=None):
+    @cherrypy.tools.json_in()
+    def DELETE(self, description=None, status=None):
         """
         Handles the DELETE request and returns a JSON response.
         :param record_id: Id of a specific todo record resource.
         :return: The status, if the operation is successful or not, along with the record that is deleted.
         """
-        res_msg = {"status": "FAIL", "data": ""}
-        return res_msg
+        try:
+            if description:
+                deleted_items = [item for item in self.todo_list.items if item.description == description]
```

Comment:
I think if we name it to "items_to_delete", it would make it more understandable since the items are not deleted yet